easybuild/tools/filetools.py

```diff
@@ -590,6 +596,19 @@ def download_file(filename, url, path, forced=False):
 
     # use custom HTTP header
     headers = {'User-Agent': 'EasyBuild', 'Accept': '*/*'}
+
+    # permit additional or override headers via http_headers_fields option
+    # whose string value contains header-fields grammar: (see rfc7230)
+    #   header-fields   = *( header-field *(CR) LF )
+    #   header-field    = field-name ":" OWS field-value OWS
+    #   OWS             = ( optional white space )
+    # Note CR may be omitted for convenience (it is absorbed in OWS and stripped)
+    if http_header_fields is not None:
+        extraheaders = dict(hf.split(':') for hf in http_header_fields.split('\n') if hf.count(':') == 1)
+        for k, v in extraheaders.items():
+            _log.debug('Setting custom HTTP header field: %s: %s' % (k, v))
```

Comment:
Logging the custom HTTP header fields essentially leads to leaking the headers in the EasyBuild log file when using `eb --debug`, I don't think you'd want that if the headers contain secrets like tokens?

See https://github.com/easybuilders/easybuild-framework/pull/3248 where we fixed an issue very much like this.

I would error on the side of caution here and just not log the custom headers at all, and verify in the tests that a custom HTTP header that includes a (fake) secret does not pop up in a debug log file anywhere.